agent/exec/container/controller.go

```diff
@@ -196,6 +200,16 @@ func (r *controller) Wait(pctx context.Context) error {
 				// and report anyways.
 				return ErrContainerDestroyed
 			}
+		case <-healthCheckTicker.C:
+			status, err := r.ContainerStatus(ctx)
+			if err == nil && status.Health.Status == "unhealthy" {
+				// in this case, we stop the container and report unhealthy status
+				if err := r.Shutdown(ctx); err != nil {
+					return errors.Wrap(err, "detect unhealthy container, shutdown failed")
+				}
```

Comment:
When detecting health check failure, container will be shut down. I don't know whether this is the correct behavior, (and the correct implementation).